proposals/fips/fleet-server-fips.md

```diff
@@ -0,0 +1,62 @@
+
+Fleet uses cryptographic operations in the following features:
+- Fleet uses TLS as a client for the following features:
+    - Fleet can conect to MySQL and Redis using TLS.
+    - Vulnerability processing uses TLS client to retrieve data from different sources.
+    - Automations: Webhooks, JIRA and Zendesk integrations make use of a TLS client.
+    - Single Sign On uses TLS to retrieve IdP metadata (from the configured MetadataURL).
+    - MDM functionality connects via TLS to Apple servers (DEP enrollment configuration).
+    - Fleet uses TLS to stream logs to many data stream systems (Kafka, Lambda, Kinesis, Firehose, etc.).
+- Fleet can act as a TLS server (can terminate TLS).
+- User authentication: Fleet uses password hashing to store and authenticate user credentials. 
+- The Single Sign On (SSO) feature in Fleet performs cryptographic verification of `SAMLResponse`s signatures. Fleet uses [goxmldsig](https://github.com/russellhaering/goxmldsig) which uses golang's stdlib crypto (`crypto/x509`, `crypto/rsa`, `crypto/sha256`, `crypto/tls` which are all compliant primitives)
+- Fleet uses `crypto/rand` to generate secrets for authentication ("session tokens" and "node keys" which are used for authenticating users and devices respectively).
+- MDM functionality (for Windows and macOS) makes use of cryptographic operations (e.g. Fleet acts as a SCEP CA server for authenticating devices)
+- Fleet license check code uses `JWT` which uses `ECDSA` for public signatures (ECDSA is a FIPS-compliant primitive).
+
+All these operations (except password hashing, see [Password hashing](#password-hashing) below) make use of Go's `crypto` standard library.
+
+## Building Go with BoringSSL as cryptographic backend
+
+As of today, the recommended way to make your Go application be FIPS 140-2 compliant is to use the BoringSSL crypto backend instead of the standard library `crypto` packages.
+
+Since we use Go 1.19, we only need to build fleet with `CGO_ENABLED=1` (already the case because of our sqlite3 dependency) and with `GOEXPERIMENT=boringcrypto`. This will automatically make Fleet use the BoringSSL cryptographic primitives instead of the stdlib `crypto` implementation (without requiring any code changes).
+
+See the [POC]https://github.com/fleetdm/fleet/compare/13288-poc-fleet-fips?expand=1()).
+
+> Source: https://kupczynski.info/posts/fips-golang/
+
+## Password hashing
+
+For password hashing Fleet currently uses the [bcrypt](https://en.wikipedia.org/wiki/Bcrypt) function.
+To be FIPS 140-2 compliant, Fleet will have to use [PBKDF2](https://en.wikipedia.org/wiki/PBKDF2] instead (with SHA-256 as hashing primitive).
+
+> Source: https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-132.pdf
+
+## Summary of tasks
+
+Assumptions around the first customer using the FIPS build:
+- All MDM funtionality will be disabled/not-used.
```

Comment:
Why?
Is MDM code not compliant as part of what you mentioned above?

Reply:
It would increase the effort. Am assuming that for the first iteration (to have it ASAP) we don't need to cover MDM.

Reply:
Got it 👍 
Let's put a price tag on it as part of the full list of things to be done.
We can then recommend not to start with it.

Reply:
@zayhanlon does customer-a intend to use MDM?

Reply:
@zhumo not at this time